### PR TITLE
Handle adding/deleting folders to a workspace

### PR DESCRIPTION
**Pull request by lszomoru**

### CODE REVIEW

**Comment by sbatten** [comment]
I wonder if we should even ask if the added folder is already untrusted

**Comment by lszomoru** [comment]
That is already the case. If any of the folders that are being added are `unspecified` or `untrusted` we will prompt.

File: src/vs/workbench/contrib/workspace/browser/workspace.contribution.ts [hunk 1]
```diff
@@ -200,6 +200,32 @@ export class WorkspaceTrustRequestHandler extends Disposable implements IWorkben
...
+					// Apply changes for the added folders
+					const addedFoldersTrustState = result.confirmed ? WorkspaceTrustState.Trusted : WorkspaceTrustState.Untrusted;
+					addedFoldersTrustStateInfo.map(i => this.workspaceTrustService.setFolderTrustState(i.uri, addedFoldersTrustState));
+
+					return;
+				}
+			}
+
+			// Refresh the workspace trust state
+			this.workspaceTrustService.refreshWorkspaceTrustState();
```
Comment (sbatten) [inline src/vs/workbench/contrib/workspace/browser/workspace.contribution.ts:226]: why is this necessary?

Comment (lszomoru) [reply src/vs/workbench/contrib/workspace/browser/workspace.contribution.ts:226]: Today, we only recalculate the workspace trust state in case there are changes to the underlying model (list of trusted/untrusted folders). Assume that you have a workspace that is currently `unspecified` and you add a folder that is `untrusted`. While there is no functional difference between the two, we need to "refresh" the workspace trust state so that the editor/status bar correctly shows the trust state.

**Review by tymsai — APPROVED** [review]
